internal/kafka/internal/services/kafka.go

```diff
@@ -706,15 +706,21 @@ func (k *kafkaService) VerifyAndUpdateKafkaAdmin(ctx context.Context, kafkaReque
 		return errors.New(errors.ErrorValidation, fmt.Sprintf("Unable to update kafka: %s with strimzi version: %s", kafkaRequest.ID, kafkaRequest.DesiredStrimziVersion))
 	}
 
-	vCompOldNewIbp, eIbp := api.CompareBuildAwareSemanticVersions(kafkaRequest.ActualKafkaIBPVersion, kafkaRequest.DesiredKafkaIBPVersion)
+	currentIBPVersion := kafkaRequest.ActualKafkaIBPVersion
+
+	if currentIBPVersion == "" { // the ibp version could be empty is the Kafka have not yet been successful provisioned at first place
+		currentIBPVersion = kafkaRequest.DesiredKafkaIBPVersion // use the desired version for checks as eventually the actual version will be this one for new Kafkas
+	}
+
+	vCompOldNewIbp, eIbp := api.CompareBuildAwareSemanticVersions(currentIBPVersion, kafkaRequest.DesiredKafkaIBPVersion)
```

Comment:
Do we really need to compare the versions if there is no actual version set? Can we skip the comparison instead?

Reply:
Thanks for the review @JameelB 
The idea of falling back to using desired version was to ensure that at least the new version on the upgrade matched the previous desired one to reduce chances of it be a breaking change.

Reply:
ah that makes sense, thanks!